backend/src/bundles/users/user.service.ts

```diff
@@ -49,10 +53,48 @@ class UserService implements Service {
     }
 
     public async update(
+        query: Record<string, number>,
+        payload: UserUpdateProfileRequestDto,
+    ): Promise<UserAuthResponseDto | null> {
+        try {
+            const userId = query['id'] as number;
+            const existingUser = await this.userRepository.find({
+                id: userId,
+            });
+            if (existingUser) {
+                const updatedUserDetails: Partial<UserDetailsModel> = {};
+                for (const property of Object.keys(payload)) {
+                    const value = payload[property];
+                    if (value) {
+                        updatedUserDetails[property] = payload[property];
+                    }
+                }
```

Comment:
seems like user will be unable to remove nickname for example with this approach, since this update will be ignored.` toUpdateProfile` could solve this and simplify everything

Reply:
Nickname is stored as a username property because it was already used everywhere as a username and also to avoid such cases

Reply:
i'm talking about case where you will receive request
```
{
  "weight": 50,
  "username": null,
}
```
with payload above. This will not delete username for user, since you are checking for truthy values, same for empty string/0.
Or am I missing something?

Reply:
Oh, I thought that the empty query on fields should not remove the previous data. But as I understood it should?